### PR TITLE
Add number rows product list

### LINKED ISSUE #4325: Add the ability to control the number of rows displayed in the product list
Add the ability to control the number of rows displayed in the product list. Suggest allowing the user to display 20, 30, 50, 100 rows.

### PR DESCRIPTION
**Pull request by benekex2**
I want to merge this change because it adds ability to change displayed number of rows in lists.
Resolves #4325 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by maarcingebala** [comment]
The selected number of rows should be stored somewhere (local storage) so that when users come back to the list, they'll have the setting they've picked before. Eventually, every list will have this feature so these settings need to be specific per list view.

**Comment by django-queries** [comment]
Here is the report for 1905cbbe8ed38d70de434589d669d213ec4ad66e (mirumee/saleor @ add-number-rows-product-list)
Base comparison is cd633473089f14bbec87dad796056efbe46dd5ad.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         43	         43	             21

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	          9	          9	              2
```

</p>
</details>

File: saleor/static/dashboard-next/types.ts [hunk 1]
```diff
@@ -9,15 +9,37 @@ export interface UserError {
   message: string;
 }
 
+export interface ListSettings {
+  name: string;
+  rowNumber: number;
+}
+
+export enum Lists {
+  CATEGORY_LIST = "CATEGORY_LIST",
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/types.ts]: There's no need to define values

File: saleor/static/dashboard-next/types.ts [hunk 2]
```diff
@@ -9,15 +9,37 @@ export interface UserError {
   message: string;
 }
 
+export interface ListSettings {
+  name: string;
+  rowNumber: number;
+}
+
+export enum Lists {
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/types.ts]: I think `ListViews` would be more descriptive name

File: saleor/static/dashboard-next/types.ts [hunk 3]
```diff
@@ -9,15 +9,37 @@ export interface UserError {
...
+  VOUCHER_LIST = "VOUCHER_LIST"
+}
+
 export interface ListProps {
   disabled: boolean;
   pageInfo?: {
     hasNextPage: boolean;
     hasPreviousPage: boolean;
   };
+  listSettings?: ListSettings;
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/types.ts]: This interface is named `ListProps` and I'm not sure if naming prop `listSettings` is any more descriptive than just `settings`

File: saleor/static/dashboard-next/types.ts [hunk 4]
```diff
@@ -9,15 +9,37 @@ export interface UserError {
   message: string;
 }
 
+export interface ListSettings {
+  name: string;
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/types.ts]: I'm sure this field isn't used anywhere

File: CHANGELOG.md [hunk 5]
```diff
@@ -21,6 +21,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Order is no longer created when the payment was unsuccessful in the API - #4500 by @NyanKiyoshi
 - Fix navigation rwd - #4511 by @benekex2
 - Create general abstraction for object metadata - #4447 by @salwator
+- Add number rows product list - #4414 by @benekex2
```
Comment (dominik-zeglen) [inline CHANGELOG.md]: I don't think it's a legit sentence

Comment (maarcingebala) [inline CHANGELOG.md]:
```suggestion
- Allow selecting the number of rows displayed in dashboard's list views - #4414 by @benekex2
```

Comment (maarcingebala) [reply CHANGELOG.md]: True, added suggestion for a better one.

**Pull request merged**